src/website/interface.ts

```diff
@@ -102,8 +102,17 @@ export class WebsiteInterface {
 		}
 	}
 
-	public async getRound(tournamentId: string): Promise<number> {
-		const matches = await this.api.getMatches(tournamentId, true);
+	public async findClosedMatch(tournamentId: string, playerId: number): Promise<WebsiteMatch | undefined> {
+		const playerMatches = await this.api.getMatches(tournamentId, false, playerId);
+		const currentRound = await this.getRound(tournamentId, playerMatches);
+		const roundMatches = playerMatches.filter(m => m.round === currentRound);
+		if (roundMatches.length > 0) {
+			return roundMatches[0];
+		}
```

Comment:
You don't ever need to call `getRound` here or devise such a system to get to it because all it does is return a property of the first element.
```suggestion
		const playerMatches = await this.api.getMatches(tournamentId, false, playerId);
		if (playerMatches.length) {
			const roundMatches = playerMatches.filter(m => m.round === playerMatches[0].round);
			return roundMatches[0]; // if length === 0 then this will be undefined at runtime
		}
```